### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by tru36rit**
This PR:
• Refactors using redux

### CODE REVIEW

File: server/utils/moviesApi.js [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-
+``
 async function fetchMovies(query) {
   try {
-    if (query) {
-      const url = makeUrl(query);
-      const response = await axios.get(url);
-      if (!Array.isArray(response.data)) return null;
-      return response.data;
+    if (query === null) {
```
Comment (lucas-aragno) [inline server/utils/moviesApi.js]: I think the `makeRequest` shouldnt be inside of the conditional

File: server/tests/server.test.js [hunk 2]
```diff
@@ -19,30 +21,39 @@ describe('express server', () => {
...
   });
 
-  // test server, router,  and controller... along with external api
-  it('Should return a list of movies related to a title search', async () => {
-    const title = 'blade runner';
-    const response = await request(app).get(`/api/search/${title}`);
-    expect(response.statusCode).toEqual(200);
-    expect(response.body).not.toHaveLength(0);
-  });
+  // // test server, router,  and controller... along with external api
```
Comment (lucas-aragno) [inline server/tests/server.test.js]: can you remove/  comment in block ?

File: server/utils/moviesApi.js (hunk 1 above)
Comment (tru36rit) [reply server/utils/moviesApi.js]: That's actually what I'm working on right now... writing my async action creator 👍

File: server/tests/server.test.js (hunk 2 above)
Comment (tru36rit) [reply server/tests/server.test.js]: Done

**Pull request merged**